services/motion/builtin/builtin.go

```diff
@@ -237,7 +270,119 @@ func (ms *builtIn) MoveOnGlobe(
 	angularVelocity float64,
 	extra map[string]interface{},
 ) (bool, error) {
-	return false, ErrNotImplemented
+	operation.CancelOtherWithLabel(ctx, motionOpId)
+
+	// create a new empty framesystem which we will add our base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+
+	// build maps of input enabled resources
+	_, resources, err := ms.fsService.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// get the localizer
+	local, ok := ms.localizers[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// assert localizer as a movementSensor to get map limits
+	movementSensor, ok := local.(movementsensor.MovementSensor)
+	if !ok {
+		return false, fmt.Errorf("cannot assert local of type %T as a movementSensor", local)
+	}
+
+	currentPosition, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	currentPose := spatialmath.GeoPointToPose(currentPosition)
+
+	// convert destination into a spatialmath pose in frame with respect to 0 latitude, 0 longitude
+	dstPose := spatialmath.GeoPointToPose(destination)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, dstPose)
+
+	// convert GeoObstacles into GeometriesInFrame then into a Worldstate
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles)
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+
+	// recalculate limits with destination and current position
+	minX := math.Min(currentPose.Point().X, dstPose.Point().X)
+	maxX := math.Max(currentPose.Point().X, dstPose.Point().X)
+	minY := math.Min(currentPose.Point().Y, dstPose.Point().Y)
+	maxY := math.Max(currentPose.Point().Y, dstPose.Point().Y)
+	limits := []referenceframe.Limit{
+		{Min: minX, Max: maxX},
+		{Min: minY, Max: maxY},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+
+	// get worldstate limits
+	limits = wrldst.BoundingBox(ctx, limits)
```

Comment:
Is the idea here to use the WorldState obstacles to set a limit on how far afield a rover may move on its way to the goal?

I suggest instead making this a (configurable) multiple of the cartesian distance from start to goal.

For example, if the start/goal are 10 meters apart, then make the limits +/- 30 meters

Reply:
the naive case of using distance between start and goal breaks in this example:

![IMG_6370](https://github.com/viamrobotics/rdk/assets/46872531/08e3d766-3b11-41b0-ab28-01057cfdb55a)

Reply:
>I suggest instead making this a (configurable) multiple of the cartesian distance from start to goal.

we can always construct an obstacle such that is is not encompassed by the "multiple"

Reply:
A sufficiently large multiple of the distance, e.g. 3x as suggested above, would manage to make it around such an obstacle.

The current implementation would only allow motion from S to the centerpoint of the obstacle and will never be able to reach the goal.